### PR TITLE
Simplify docker build and reduce build image

### PR DESCRIPTION
**Pull request by yanxke**
- Merge the two docker files into one to simplify management.
- Put build process in shell script to and remove source and object files to reduce docker image.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,48 +1,15 @@
-FROM ubuntu:22.04
+ARG BASE_IMAGE=ubuntu:22.04
+
+FROM $BASE_IMAGE
 
 ARG MASTER
```
Comment (cdcseacave) [inline docker/Dockerfile:5]: master seems not reproduced, can you pass the argument to the script?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,48 +1,15 @@
...
 ARG USER_ID
 ARG GROUP_ID
```
Comment (cdcseacave) [inline docker/Dockerfile:7]: is this still needed

File: docker/README.md [hunk 3]
```diff
@@ -14,7 +14,11 @@
 
 You can also build the docker image from scratch based on the **Dockerfile** (perhaps with your own changes / modifications) using:
 
-        ./buildFromScratch.sh /path/where/your/SFM/results/are
+```sh
+./buildFromScratch.sh /path/where/your/SFM/results/are
+# With CUDA support:
+./buildFromScratch_CUDA.sh /path/where/your/SFM/results/are
```
Comment (cdcseacave) [inline docker/README.md:20]: can you remove the buildFromScratch_CUDA.sh too and add a param to the normal script?

File: docker/buildInDocker.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+fi
+
+mkdir openMVS_build
+cd openMVS_build &&\
+    cmake . ../openMVS -DCMAKE_BUILD_TYPE=Release -DVCG_ROOT=/vcglib $OPENMVS_BUILD_ARG
+
+# Install OpenMVS library
+make -j4 &&\
+    make install &&\
+    cd .. && rm -rf openMVS_build openMVS vcglib
```
Comment (cdcseacave) [inline docker/buildInDocker.sh:73]: removing the source code is fine in general I guess, but we need to make sure the install really copies everything needed: header files, libraries, and helper scripts, like the python scripts, can you pls test that?

**Review by cdcseacave — CHANGES_REQUESTED** [review]
Interesting approach, I am not a docker expert but I imagine running a script inside instead of docker commands has some disadvantages (like reusing images maybe), though I am fine with it, so I will merge it.

But before I do, pls address my comments and change the PR base to develop branch.

**Comment by cdcseacave** [comment]
also pls synchronize with @satyajitghana which also changed this files, see #954

File: docker/buildInDocker.sh (hunk 4 above)
Comment (yanxke) [reply docker/buildInDocker.sh:73]: Are there test suites I can run? I'm not familiar with the structure of MVS to reliably test this manually.

**Comment by yanxke** [comment]
> Interesting approach, I am not a docker expert but I imagine running a script inside instead of docker commands has some disadvantages (like reusing images maybe), though I am fine with it, so I will merge it.
> 
> But before I do, pls address my comments and change the PR base to develop branch.

The disadvantage of running the script inside the Docker container is that you can't use the partially built image and resume from there if the script fails in the middle. Since we don't expect the script to fail often, and the original build script explicitly disables the build cache, this is not an issue.

File: docker/README.md (hunk 3 above)
Comment (yanxke) [reply docker/README.md]: Fixed

File: docker/Dockerfile (hunk 2 above)
Comment (yanxke) [reply docker/Dockerfile:7]: Fixed by passing in parameters.

File: docker/Dockerfile (hunk 1 above)
Comment (yanxke) [reply docker/Dockerfile]: Fixed

**Comment by yanxke** [comment]
Changed target branch to develop.

**Pull request merged**